File: src/main/java/com/machfour/macros/ingredients/IngredientSpec.java
package com.machfour.macros.ingredients;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

class IngredientSpec {
    @NotNull
    final String indexName;
    @NotNull
    final Double quantity;
    @NotNull
    final String quantityUnit;
    @Nullable
    final String notes;

    IngredientSpec(@NotNull String indexName, @NotNull Double quantity, @NotNull String quantityUnit, @Nullable String notes) {
        this.indexName = indexName;
        this.quantity = quantity;
        this.quantityUnit = quantityUnit;
        this.notes = notes;
    }

    @Override
    public String toString() {
        return String.format("{ indexName: %s, quantity: %.2f, quantityUnit: %s, notes: %s }",
                indexName, quantity, quantityUnit, notes);
    }
}

File: libmacros/src/main/kotlin/com/machfour/macros/ingredients/IngredientSpec.kt
package com.machfour.macros.ingredients

import com.machfour.macros.formatting.toString

data class IngredientSpec(val indexName: String, val quantity: Double, val unit: String, val notes: String?) {
    override fun toString(): String {
        return "{ indexName: $indexName, quantity: ${quantity.toString(2)}, quantityUnit: $unit, notes: $notes }"
    }
}